python/cudf/cudf/core/indexed_frame.py

```diff
@@ -3476,22 +3496,32 @@ def _explode(self, explode_column: Any, ignore_index: bool):
             idx = None if ignore_index else self._index.copy(deep=True)
             return self.__class__._from_data(data, index=idx)
 
-        explode_column_num = self._column_names.index(explode_column)
+        column_index = self._column_names.index(explode_column)
         if not ignore_index and self._index is not None:
-            explode_column_num += self._index.nlevels
+            index_offset = self._index.nlevels
+        else:
+            index_offset = 0
 
         exploded = libcudf.lists.explode_outer(
             [
                 *(self._index._data.columns if not ignore_index else ()),
                 *self._columns,
             ],
-            explode_column_num,
+            column_index + index_offset,
         )
-
+        # We must copy inner datatype of the exploded list column to
+        # maintain struct dtype key names
+        exploded_dtype = cast(
+            ListDtype, self._columns[column_index].dtype
+        ).element_type
         return self._from_columns_like_self(
             exploded,
             self._column_names,
             self._index_names if not ignore_index else None,
+            override_dtypes=(
+                exploded_dtype if i == column_index else None
+                for i, _ in enumerate(self._columns)
```

Comment:
No need to fetch the column accessors if they're unused.
```suggestion
                for i in range(len(self._columns))
```